### ISSUE TITLE
Characters being reused for multiple "words" in vertical Japanese text when OCRing three or more lines at once with LSTM "best" models

### ISSUE THREAD

**Issue by wareya** [tesseract-ocr/tesseract#1117]
### Environment

Using the training data from https://github.com/tesseract-ocr/tessdata/tree/master/best on tesseract 4

Exact command line parameters (aside from input, output, and config) are:

    -l jpn_vert+jpn --psm 5 --oem 1

* **Tesseract Version**: 

    tesseract 4.00.00alpha
     leptonica-1.74.4 (Jun 22 2017, 16:20:35) [MSC v.1900 DLL Release x86]
      libjpeg 9b : libpng 1.6.28 : libtiff 4.0.7 : zlib 1.2.11 : libwebp 0.6.0 : libopenjp2 2.1.2

* **Commit Number**: 

Specific build ripped from the 2017-09-04 build of VietOCR-5.0-alpha, however the same exact behavior was experienced with "zip file with cppan generated .dll and .exe files" from https://github.com/tesseract-ocr/tesseract/wiki/4.0-with-LSTM

* **Platform**: 

Windows 7 x64

### Current Behavior:

https://i.imgur.com/rtzROTu.png

OCRs correctly with the following config:

    preserve_interword_spaces 1
    paragraph_text_based 0
    textord_old_baselines 0

OCRs wrong with the following config:

    preserve_interword_spaces 1
    paragraph_text_based 0
    textord_old_baselines 1

    ケータイは
    カバパバンに
    入れてあるし

カバパバン should be カバン

https://i.imgur.com/TA1Qa2O.png

OCRs correctly with the following config:

    preserve_interword_spaces 1
    paragraph_text_based 0
    textord_old_baselines 1

OCRs wrong with the following config:

    preserve_interword_spaces 1
    paragraph_text_based 0
    textord_old_baselines 0

    ティイツシュは
    スカートの
    ポケットだけら
    大丈夫!

ティイツシュ should be ティッシュ (or at least テイツシユ with set of small kana)

If preserve_interword_spaces is set to 0, it appears that the relevant characters are being used more than once for different "words":

ケー タイ は
カバ パ バン に
入れ て ある し

ティ イツ シュ は
スカ ー ト の
ポケ ッ ト だ けら
大 丈夫 !

**Regardless of configuration, both examples OCR correctly if OCRed one or two lines at a time, but start OCRing wrong if three or more lines are OCRed at once.**

psm settings other than 5 (or 6 for horizontal panels) give results like "empty page", "not enough words", or missing characters/words entirely, so I can't compare them.

### Expected Behavior:

With at least one of the configs, for both images to OCR correctly, without characters being used in multiple words.

### Suggested Fix:

N/A

**Comment by wareya** [comment]
The カバン example in the original post is alternatively fixed by setting "lstm_use_matrix 0". It seems like "lstm_use_matrix 1" is very bad.

More examples with bad OCR with various combinations of these two settings below. Summary: lstm_use_matrix 0 is universally better than 1, textord_old_baselines 0 is better than 1 most of the time but can throw away entire lines.

Single line: https://i.imgur.com/BTeN4Lx.png

OCRs with "reused characters" regardless of the value of textord_old_baselines or lstm_use_matrix. A fictitious ょ is inserted no matter what. Some other croppings of this particular are improved by setting "lstm_use_matrix 0", but it doesn't always help.

Setting both "lstm_use_matrix 0" and "textord_old_baselines 0" seems to help in general, but doesn't fix the problem in the original post. Example:

https://i.imgur.com/ji4CQdr.png

Config:

    preserve_interword_spaces 0
    paragraph_text_based 0
    textord_old_baselines 1
    lstm_use_matrix 1

Output:

    ほ の り ちゃ ん が
    一 緒 の 部 活 に

    入っ て くれ て

    うれ し いね は っ て

fictitious は

Config:

    preserve_interword_spaces 0
    paragraph_text_based 0
    textord_old_baselines 1
    lstm_use_matrix 0

Output:

    ほ の り ち ゃ ん が
    一 緒 の 部 活 に

    入 っ て く れ て

    う れ し い ね は っ て

fictitious は, still

Config:

    preserve_interword_spaces 0
    paragraph_text_based 0
    textord_old_baselines 0
    lstm_use_matrix 1

Output:

    ほ ば のり ちゃ ん が
    一 緒 の 部 活 に

    入っ て くれ て

    うれ し いね っ て

fictitious ば



Config:

    preserve_interword_spaces 0
    paragraph_text_based 0
    textord_old_baselines 0
    lstm_use_matrix 0

Output:

    ほ の り ち ゃ ん が
    一 緒 の 部 活 に

    入 っ て く れ て

    う れ し い ね っ て

Correct

Setting "textord_old_baselines 0" is definitely not ideal, because 1 (the default) is responsible for accuracy gains in various cases:

https://i.imgur.com/tH8yx84.png

textord_old_baselines 1 -> 部 長 な ん だ か ら
textord_old_baselines 0 -> 部 長 な ん だ ヵ か ら

Image where "textord_old_baselines 0" throws away an entire line: and "lstm_use_matrix 1" causes character misrecognition: https://i.imgur.com/8d65wSf.png

textord_old_baselines 1, lstm_use_matrix 1 -> ちっ DI を で　　心配 だ し　　と りあ え ず 入っ て

textord_old_baselines 0, lstm_use_matrix 1 -> めい 子 ち ゃ ん カ　　と りあ え ず 入っ て

textord_old_baselines 1, lstm_use_matrix 0 -> め じ 子 ち ゃ ん カ　　心 配 だ し　　と り あ え ず 入 っ て

textord_old_baselines 0, lstm_use_matrix 0 -> め い 子 ち ゃ ん カ　　と り あ え ず 入 っ て

(the above double fullwidth spaces are substitutes for the newlines)

**Comment by amitdo** [comment]
What's the content of the official best traineddata's config file?

**Comment by wareya** [comment]
AFAIK, the stuff under https://github.com/tesseract-ocr/tessdata/tree/master/best is all there is for the "best" traineddata.

**Comment by amitdo** [comment]
Yes, but there's a way to unpack the traineddata and see its individual files.

**Comment by wareya** [comment]
[jpn.config.txt](https://github.com/tesseract-ocr/tesseract/files/1291681/jpn.config.txt)
[jpn_vert.config.txt](https://github.com/tesseract-ocr/tesseract/files/1291683/jpn_vert.config.txt)

**Comment by hoangtocdo90** [comment]
@wareya thanks . I have same question . In a new best jpn.traineddata some times result text having double of char.  .
 example input : ABCDEFG -> output :ABCDEFFG.
Not many a little only but i can't understand what's wrong with new jpn.traineddata. I don't know but i think in the previous version of best traineddata [4.0] this bug not appear 
https://github.com/tesseract-ocr/tessdata/tree/4.00

**Comment by Shreeshrii** [comment]
Similar issue - https://github.com/tesseract-ocr/tesseract/issues/1060
